k2/csrc/test_utils.cu

```diff
@@ -0,0 +1,74 @@
+/**
+ * @brief
+ * test_utils
+ *
+ * @copyright
+ * Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include <algorithm>
+#include <cstdlib>
+#include <vector>
+
+#include "k2/csrc/fsa.h"
+#include "k2/csrc/host/fsa_util.h"
+#include "k2/csrc/host_shim.h"
+#include "k2/csrc/test_utils.h"
+
+namespace k2 {
+
+void ToNotTopSorted(Fsa *fsa) {
+  K2_CHECK_EQ(fsa->Context()->GetDeviceType(), kCpu);
+
+  int32_t num_states = fsa->TotSize(0);
+  std::vector<int32_t> order(num_states);
+  std::iota(order.begin(), order.end(), 0);
+  std::random_shuffle(order.begin() + 1, order.end() - 1);
+
+  Array1<Arc> &arcs = fsa->values;
+  Arc *arcs_data = arcs.Data();
+  int32_t num_arcs = arcs.Dim();
+  for (int32_t i = 0; i != num_arcs; ++i) {
+    int32_t src_state = arcs_data[i].src_state;
+    int32_t dest_state = arcs_data[i].dest_state;
+    arcs_data[i].src_state = order[src_state];
+    arcs_data[i].dest_state = order[dest_state];
+  }
+
+  auto lambda_comp = [](const Arc &a, const Arc &b) -> bool {
+    return a.src_state < b.src_state;
+  };
+  std::sort(arcs_data, arcs_data + num_arcs, lambda_comp);
+  for (int32_t i = 0; i != num_arcs; ++i) {
+    arcs_data[i].score = i;
+  }
+
+  bool error = true;
+  *fsa = FsaFromArray1(arcs, &error);
+  K2_CHECK(!error);
+}
+
+Fsa GetRandFsa() {
+  k2host::RandFsaOptions opts;
+  opts.num_syms = 5 + rand() % 100;
+  opts.num_states = 10 + rand() % 2000;
```

Comment:
Again, I still suggest we can go to the approach of `RandomRaggedShape` I metioned above to generate random Fsas, it can generate reasonable random Fsas (and empty Fsas as well, which I believe we need to test some cornner cases according to me debug experience). Last but not least, the code will be much simpler (the only new code is just to generate arcs according fsa meta info ) and clearer (no RandomGenerator in `host`.)